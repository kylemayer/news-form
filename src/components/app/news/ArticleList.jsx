import React from 'react';
import PropTypes from 'prop-types';
import Articles from './Article';

const ArticleList = ({ articles }) => (
  <ul aria-label="article-list">
    {articles.map(({ title, author, description }) => (
      <li key={title}>
        <Articles
          title={title}
          author={author}
          description={description}
        />
      </li>
    ))}
  </ul>
);

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
    })
  ).isRequired,
};

export default ArticleList;
