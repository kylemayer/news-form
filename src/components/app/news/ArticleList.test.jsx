import dotenv from 'dotenv';
dotenv.config();

import React from 'react';
import { render, screen } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList file', () => {
  it('should display article in a ul', async () => {
    const articles = [
      {
        title: 'iPhone 13 Release',
        author: 'Sam',
        description: 'Five new colors',
      },
      {
        title: 'PS5 on Sale',
        author: 'Sally',
        description: 'Sale on Friday',
      },
    ];

    render(
      <ArticleList articles={articles} />
    );

    const ul = await screen.findByRole('list', { name: 'article' });
    expect(ul).toMatchSnapshot();
  });
});
