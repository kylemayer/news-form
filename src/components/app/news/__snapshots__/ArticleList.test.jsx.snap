// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList file should display article in a ul 1`] = `
<ul
  aria-label="article"
>
  <li>
    <div>
      <h4>
        iPhone 13 Release
      </h4>
      <p>
        Sam
      </p>
      <p>
        Five new colors
      </p>
    </div>
  </li>
  <li>
    <div>
      <h4>
        PS5 on Sale
      </h4>
      <p>
        Sally
      </p>
      <p>
        Sale on Friday
      </p>
    </div>
  </li>
</ul>
`;
